packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -1910,22 +1906,18 @@ function updateSuspenseComponent(current, workInProgress, renderLanes) {
   if (current === null) {
     // Initial mount
     // If we're currently hydrating, try to hydrate this boundary.
-    // But only if this has a fallback.
```

Comment:
This change is reflected in the ReactDOMServerPartialHydration-test.internal.js test.